packages/helpers/nextjs.tsx

```diff
@@ -0,0 +1,6 @@
+import { GetServerSidePropsContext } from 'next';
+
+export function urlFromContext(context: GetServerSidePropsContext): string {
+   const protocol = context.req.headers.referer?.split('://')[0] || 'https';
+   return `${protocol}://${context.req.headers.host}${context.resolvedUrl}`;
+}
```

Comment:
You could use the actual URL object. I know you may toString it later, but the class may be nice down the road

```suggestion
export function urlFromContext(context: GetServerSidePropsContext): URL {
   const protocol = context.req.headers.referer?.split('://')[0] || 'https';
   return new URL(`${protocol}://${context.req.headers.host}${context.resolvedUrl}`);
}
```

Reply:
I agree, but we use string urls throughout most of this app unfortunately, so this would be a more siginificant refactor if we were to do it right. I'd rather leave it to another pull.